storage/store.go

```diff
@@ -833,6 +835,78 @@ func (s *Store) MergeRange(subsumingRng *Range, updatedEndKey proto.Key, subsume
 	return subsumedRng, nil
 }
 
+func (s *Store) removeAllMetadata(rep *Replica) error {
+	rangeID := rep.Desc().RangeID
+
+	if err := engine.MVCCDelete(s.engine, nil, keys.RangeGCMetadataKey(rangeID), 
+		s.Clock().Now(), nil); err != nil{
+		return util.Errorf("cannot delete gdmeta %s", keys.RangeGCMetadataKey(rangeID))
+	}
+
+	if err := engine.MVCCDelete(s.engine, nil, keys.RangeLastVerificationTimestampKey(rangeID), 
+		s.Clock().Now(), nil); err != nil {
+		return util.Errorf("cannot delete last ts %s", keys.RangeLastVerificationTimestampKey(rangeID))
+	}
+
+	// RaftLogKey 
+	lastIndex, err := rep.LastIndex()
+	if err != nil {
+		return err
+	}
+	// Delete any previously appended log entries which never committed.
+	for i := uint64(0); i <= lastIndex; i++ {
+		if err := engine.MVCCDelete(s.engine, nil, keys.RaftLogKey(rangeID, i), s.Clock().Now(), nil); err != nil {
+			return util.Errorf("cannot delete log %s", keys.RaftLogKey(rangeID, i))
+		}
+	}
+
+	// ResponseCacheKey
+	if err = engine.MVCCDelete(s.engine, nil, keys.ResponseCacheKey(rangeID, nil), s.Clock().Now(), nil); err != nil {
```

Comment:
there is a lot of duplication here, which you can eliminate using a slice of anonymous structs:
```go
now := s.Clock().Now()
metadata := []struct{
  name string
  key    proto.Key
}{
  {"cache", keys.ResponseCacheKey(rangeID, nil)},
  {"logprefix", keys.RaftLogPrefix(rangeID)},
  {"hardstate", keys.RaftHardStateKey(rangeID)},
 ...
}
for _, metadatum := range metadata {
  if err := engine.MVCCDelete(s.engine, nil, metadatum.key, now, nil); err != nil {
    return util.Errorf("cannot delete %s key %s: %s", metadatum.name, metadatum.key, err)
  }
}
```